team/bundles/org.eclipse.compare.core/src/org/eclipse/compare/rangedifferencer/RangeDifferencer.java

```diff
@@ -152,6 +152,7 @@ else if (diff.leftLength() == 0) {
 					optimized = factory.createRangeDifference(diff.kind(), diff.rightStart - moved, diff.rightLength, diff.leftStart - moved, diff.leftLength);
 				}
 			}
+			previousEnd = Math.max(optimized.rightEnd(), optimized.leftEnd());
```

Comment:
I am not 100% sure here, do we need two variables `previousLeftEnd` / `previousRightEnd` instead?

Reply:
It looks okay to me, but I have been thinking back and forth for 10 minutes now, too. When I wrote that code, it looked all so clear, and now it doesn't anymore...

Reply:
Hm, I can't get my head around it either.

I also have some more concerns, see #66

Reply:
I'm okay with reverting.